src/apps/common/ns_turn_utils.c

```diff
@@ -546,8 +584,28 @@ void turn_log_func_default(TURN_LOG_LEVEL level, const char* format, ...)
 	} else {
 		so_far += snprintf(s, sizeof(s), "%lu: ", (unsigned long)log_time());
 	}
-	so_far += snprintf(s + so_far, sizeof(s)-100, (level == TURN_LOG_LEVEL_ERROR) ? ": ERROR: " : ": ");
-	so_far += vsnprintf(s + so_far,sizeof(s) - (so_far+1), format, args);
+	if (_log_file_line_set)
+        so_far += snprintf(s + so_far, MAX_RTPPRINTF_BUFFER_SIZE - (so_far + 1), "%s(%d):", file, line);
```

Comment:
Like it!
But I would keep it out of this PR and also make it configurable (turn on/off detailed logging)
If that's a lot of work it can be done in subsequent PR

Reply:
Agree. but need it now. you can delete it after add third logging.